assets/src/edit-story/app/media/local/typedefs.js

```diff
@@ -0,0 +1,141 @@
+ * @property {SetMediaTypeFn} setMediaType action to set media type
+ * @property {SetNextPageFn} setNextPage action to set next page
+ * @property {SetSearchTermFn} setSearchTerm action to set search term
+ * @property {UpdateMediaElementFn} updateMediaElement action update media
+ * @property {UploadMediaFn} uploadMedia action to upload media
+ * @property {UploadVideoPosterFn} uploadVideoPoster action upload video poster
+ */
+
+/**
+ * 'local media' state object exposed to the rest of application through context
+ * providers.
+ *
+ * @typedef {Object} LocalMediaContextState
+ * @property {boolean} hasMore is there more media
+ * @property {boolean} isMediaLoaded is media loaded
+ * @property {boolean} isMediaLoading is media loading
+ * @property {boolean} isUploading is media uploading
+ * @property {Array.<Media>} media array of media
+ * @property {string} mediaType current media type filter selected
+ * @property {string} nextPageToken the page token for the next page
+ * @property {string} pageToken page token
+ * @property {Array.<number>} processed videos that have been processed
+ * @property {Array.<number>} processing videos that haven't been processed
+ * @property {string} searchTerm search term
+ * @property {number} totalPages total pages
+ */
+
+/**
+ * 'local media' context exposed to application through context providers containing
+ * state and action.
+ *
+ * @typedef {Object} LocalMediaContext
+ * @property {LocalMediaContextState} state state of a single provider
+ * @property {LocalMediaContextActions} actions actions for the provider
+ */
+
+/**
+ * 'local media' state object used internally by the reducer.
+ *
+ * @typedef {Object} LocalMediaReducerState
+ * @property {boolean} hasMore is there more media
+ * @property {boolean} isMediaLoaded is media loaded
+ * @property {boolean} isMediaLoading is media loading
+ * @property {Array.<Media>} media array of media
+ * @property {string} mediaType current media type filter selected
+ * @property {string} nextPageToken the page token for the next page
+ * @property {string} pageToken page token
+ * @property {Array.<number>} processed ids of videos that have been processed
+ * @property {Array.<number>} processing ids of videos that haven't been
+ * processed
+ * @property {string} searchTerm search term
+ * @property {number} totalPages total pages
+ */
+
+/**
+ * 'local media' actions used internally by the reducer.
+ *
+ * @typedef {Object} LocalMediaContextActions
+ * @property {DeleteMediaElementFn} deleteMediaElement action to delete a media
+ * @property {FetchMediaErrorFn} fetchMediaError action when fetching media has
+ * error
+ * @property {FetchMediaStartFn} fetchMediaStart action when fetching media
+ * starts
+ * @property {FetchMediaSuccessFn} fetchMediaSucccess action when fetching media
+ * is successful
+ * @property {ResetFiltersFn} resetFilters action to reset filters ?
+ * @property {ResetWithFetchFn} resetWithFetch action to reset with fetch ?
```

Comment:
Remove the '?' at the end of the sentence.
resetWithFetch resets the filter in the state, and then refetches the media from the server.
This fn will be hidden as a part of https://github.com/google/web-stories-wp/issues/2152.

Reply:
Done.